src/vm.c

```diff
@@ -2748,15 +2748,9 @@ mrb_vm_exec(mrb_state *mrb, const struct RProc *proc, const mrb_code *pc)
       NEXT;
     }
 
-    CASE(OP_ARYDUP, B) {
-      mrb_value ary = regs[a];
-      if (mrb_array_p(ary)) {
-        ary = mrb_ary_new_from_values(mrb, RARRAY_LEN(ary), RARRAY_PTR(ary));
-      }
-      else {
-        ary = mrb_ary_new_from_values(mrb, 1, &ary);
-      }
-      regs[a] = ary;
+    CASE(OP_ARYSPLAT, B) {
+      regs[a] = mrb_ary_splat(mrb, regs[a]);
```

Comment:
Can you please make assignments to `regs` via temporary variables?
See commit 8981b2fb28dbaed76078aff6c84f19b90832aaaf.
\# 🐈‍⬛  I tend to forget too.

Reply:
No, we have no problem using them if necessary.
However, we need to separate statements when we make assignments to `regs`, and function or method calls that involve GC occurrences in a single statement.

In this case, for example, it is better to do the following:

```c
mrb_value ary = mrb_ary_splat(mrb, regs[a]);
regs[a] = ary;
```